Automattic/wp-calypso
# Fixing new google analytics card design to work properly with atomic

#### Changes proposed in this Pull Request

* This PR fixes the redesigned Google Analytics settings card (https://github.com/Automattic/wp-calypso/pull/49629) to adjust its behavior for atomic sites on WPCOM.
* Atomic sites differ slightly from simple sites in their handling of the Google Analytics plugin; it is a Jetpack module which needs to be actively enabled or disabled, rather than simply setting or removing the tracking code value. 
* Additionally, atomic sites offer a toggle on the settings card for enabling or disabling the option to anonymize IPs when tracking analytics.

(A/N - originally all Jetpack and non-Jetpack code for this card were in the same component; with the switch to a card design where the form is shown and hidden based on a toggle, the state management for handling both these cases at once got pretty gnarly, hence the approach in this PR which splits them into a Jetpack and non-Jetpack component. Open to suggestions if another approach seems preferable but I think this works cleanly enough.)

#### Testing instructions



* If necessary, create a new WPCOM site with Business or Ecommerce plan and a plugin enabled to make it Atomic.
* Navigate to `Tools->Marketing->Traffic`.
* Verify that you can enable and disable Google Analytics, enter a tracking code, and see it reflected in your site's header code.
* Verify that you can perform the same actions with a simple site, but that the form appears differently.

Simple site:
![image](https://user-images.githubusercontent.com/13437011/109065567-67575e00-76b1-11eb-87e4-de0dbf1f6ea6.png)


Atomic site:
![image](https://user-images.githubusercontent.com/13437011/109065518-51e23400-76b1-11eb-9d28-235788a4b423.png)



Related to p1614032820004200-slack-C01HV13S295
